linera-service/src/cli_wrappers/local_kubernetes_net.rs

```diff
@@ -0,0 +1,196 @@
+            self.testing_prng_seed,
+            self.binaries_dir,
+            String::from("linera-test-local"),
+        )?;
+        let client = net.make_client();
+        net.run().await.unwrap();
+        Ok((net, client))
+    }
+}
+
+#[async_trait]
+impl LineraNet for LocalKubernetesNet {
+    async fn ensure_is_running(&mut self) -> Result<()> {
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::namespaced(client, "default");
+
+        let lp = ListParams::default().labels("app=validator-1");
+        for p in pods.list(&lp).await? {
+            if let Some(status) = p.status {
+                if let Some(phase) = status.phase {
+                    if phase != "Running" {
+                        bail!(
+                            "Validator {} is not Running",
+                            p.metadata.name.expect("Fetching pod name should not fail")
+                        );
+                    }
+                }
+            }
+        }
+
+        let lp = ListParams::default().labels("app=shards");
+        for p in pods.list(&lp).await? {
+            if let Some(status) = p.status {
+                if let Some(phase) = status.phase {
+                    if phase != "Running" {
+                        bail!(
+                            "Shard {} is not Running",
+                            p.metadata.name.expect("Fetching pod name should not fail")
+                        );
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn make_client(&mut self) -> ClientWrapper {
+        let client = ClientWrapper::new(
+            self.tmp_dir.clone(),
+            self.network,
+            self.testing_prng_seed,
+            self.next_client_id,
+        );
+        if let Some(seed) = self.testing_prng_seed {
+            self.testing_prng_seed = Some(seed + 1);
+        }
+        self.next_client_id += 1;
+        client
+    }
+
+    async fn terminate(mut self) -> Result<()> {
+        if let Some(mut port_forward_process) = self.port_forward_process {
+            port_forward_process.kill().await?;
+        }
+
+        let cluster_id = self.cluster_id.expect("Cluster ID should be set");
+        KindWrapper::delete(cluster_id).await?;
+        DockerWrapper::rm(&self.container_name).await?;
+        Ok(())
+    }
+}
+
+impl LocalKubernetesNet {
+    fn new(
+        network: Network,
+        testing_prng_seed: Option<u64>,
+        binaries_dir: Option<String>,
+        container_name: String,
+    ) -> Result<Self> {
+        Ok(Self {
+            network,
+            testing_prng_seed,
+            next_client_id: 0,
+            tmp_dir: Arc::new(tempdir()?),
+            binaries_dir,
+            port_forward_process: None,
+            container_name,
+            cluster_id: None,
+        })
+    }
+
+    async fn run(&mut self) -> Result<()> {
+        self.cluster_id = Some(get_random_number());
```

Comment:
Is there no other way to get fresh cluster id?

Reply:
We can set the ID to be whatever we want, so I just decided to set it to some numerical ID. Not sure if there's a better way, open to suggestions :)

Reply:
Ask the user then?